query-parser/src/lib.rs

```diff
@@ -8,6 +8,11 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+#![allow(unused_imports)]
```

Comment:
This doesn't really help with anything with the unused import warning. Macros need to be imported at the crate root it seems, and the crate root ("lib.rs") doesn't use the macro directly.

Importing it in the tests module isn't allowed.